### ISSUE TITLE
Superfish menu block still renders when all of the menu items are not enabled

### ISSUE THREAD

**Issue by bspeare** [uiowa/uiowa#8421]
# Problem
Brandon shared this report: I came across an issue with the site https://mlk.uiowa.edu/. Because this is a one page site, it doesn’t have any main navigation items. I’m getting an accessibility error in Siteimprove because the superfish unordered list is empty. [https://help.siteimprove.com/support/solutions/articles/80001051793-sia-r68-container-element-is-empty-explained](https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.siteimprove.com%2Fsupport%2Fsolutions%2Farticles%2F80001051793-sia-r68-container-element-is-empty-explained&data=05%7C02%7Cits-web%40uiowa.edu%7C131ee0e7d5654d7c7dc208dd3672c25f%7C1bc445959aba4fc3b8ec7b94a5586fdc%7C1%7C0%7C638726588697158957%7CUnknown%7CTWFpbGZsb3d8eyJFbXB0eU1hcGkiOnRydWUsIlYiOiIwLjAuMDAwMCIsIlAiOiJXaW4zMiIsIkFOIjoiTWFpbCIsIldUIjoyfQ%3D%3D%7C0%7C%7C%7C&sdata=qpGK14wMre%2Fo0hkpO5OE%2FBZ%2BlmgBzvWiVLCdheyFvZg%3D&reserved=0)

# Steps to reproduce

- Go to https://mlk.uiowa.edu/. 

# Proposed solution

**Comment by quamsta** [comment]
@bspeare 

This also may be occurring when there are unpublished pages in a menu as well. We are working on moving some content under "Education -> Fellowships" in https://pathology.medicine.uiowa.edu that isn't published yet and I noticed the same SiteImprove errors for an empty container

<img width="1606" alt="Image" src="https://github.com/user-attachments/assets/e2f51f0d-3c8f-4f96-9676-93526b71dc3a" />

**Comment by joewhitsitt** [comment]
I think this is intersecting with a couple other issues. https://github.com/uiowa/uiowa/issues/7970 and https://github.com/uiowa/uiowa/issues/3406

Ultimately, if we continue the whack-a-mole with superfish, let me resurface this one: https://github.com/uiowa/uiowa/issues/4973

**Comment by bspeare** [comment]
The empty nav can also been seen at https://signage.sites.uiowa.edu/ with the gray border appearing under the gold bar. 

<img width="1554" height="220" alt="Image" src="https://github.com/user-attachments/assets/cb618731-744c-46c0-85ff-b8e64473486c" />

**Comment by bspeare** [comment]
Tested this on simplytest.me with superfish and Olivero theme and it is an issue there.  The empty `<ul>` doesn't happen with the built-in Olivero dropdown so this appears unique to superfish.  It's possible that https://www.drupal.org/project/superfish/issues/3048722 might fix this but I'm not seeing a specific issue for this.  The workaround for the empty `<ul>` for the MLK site is to delete the unpublished menu items.  Because this appears unique to superfish, it seems like we might want to solve this by reworking https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/uids_base.theme#L1383 and filing an upstream issue for superfish. 

How to recreate:
- Publish a site with all menu items disabled and view the page source for empty `<ul>`
- Delete disabled menu items and view page source and see if empty `<ul>` is rendered. 
- Workaround doesn't solve the issue of the empty nav and divs being rendered.   Reworking https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/uids_base.theme#L1383 will solve that part.
